src/main/java/seedu/address/model/person/Person.java

```diff
@@ -57,7 +57,7 @@ public boolean isSameElement(UniqueElement otherElement) {
         return otherPerson != null
                 && otherPerson.getName().equals(getName())
                 && (otherPerson.getDateOfBirth().equals(getDateOfBirth())
-                || otherPerson.getGender().equals(getGender()));
+                && otherPerson.getGender().equals(getGender()));
```

Comment:
can you edit this. a Person is identified by their name only so this should be the return value:

return otherPerson != null
                && otherPerson.getName().equals(getName())